### PR TITLE
Read current site without using RegEx

### PR DESCRIPTION
**Pull request by exacs**
This PR changes the way kpm guesses which is the current "site" (external, student-web or intranet).

Instead of looking at the URL, it looks if a given class is present in the header.

### CODE REVIEW

**Comment by kaj** [comment]
How about canvas and other apps that don't (yet) have the new kthstyle header?  I guess they will just be "none of the above", and that is probably fine?

(I just want that to be a concious decision, rather than just happening)

**Comment by exacs** [comment]
@kaj Yes, Canvas and sites without any of the headers will be "none". All links will be "default" (blue text, transparent background), none of them marked.

File: kpm/kpm-frontend/src/components/entrances.tsx [hunk 1]
```diff
@@ -14,15 +10,11 @@ const labels = {
...
-  if (STUDENT_REGEX2.test(currentPage)) {
-    return "student";
-  } else if (INTRA_REGEX.test(currentPage)) {
-    return "intra";
-  } else if (EXTERNAL_REGEX.test(currentPage)) {
-    return "external";
-  }
+  return SITES.find((site) =>
+    document.querySelector("kth-header")?.classList.contains(site)
+  );
```
Comment (jhsware) [inline kpm/kpm-frontend/src/components/entrances.tsx:17]: Do we have a custom element called "kth-header" or is this query selector a typo?

File: kpm/kpm-frontend/src/components/entrances.tsx [hunk 2]
```diff
@@ -14,15 +10,11 @@ const labels = {
 
 /** Get the current site */
 function getCurrentSite(): "external" | "intra" | "student" | undefined {
-  const currentPage = window.location.toString();
+  const SITES = ["external", "intra", "student"] as const;
```
Comment (jhsware) [inline kpm/kpm-frontend/src/components/entrances.tsx:13]: The site names don't match the classnames in the variable links. Considering the check classList.contains(site) this feels strange that they are so close but still not the same. If it is intentional, perhaps one could add a comment.

File: kpm/kpm-frontend/src/components/entrances.tsx (hunk 1 above)
Comment (jhsware) [reply kpm/kpm-frontend/src/components/entrances.tsx:17]: notLoggedIn.js:L52 indicates typo.

File: kpm/kpm-backend/src/widget.js/notLoggedIn.js [hunk 3]
```diff
@@ -48,40 +49,29 @@
...
     .join("");
 
   let login = `<a class="kth-menu-item kpm-login" href="${url}?nextUrl=${location.href}">${lbls.l[lang]}</a>`;
-
-  let currentLabel =
-    current === "" ? links.external.label[lang] : links[current].label[lang];
+  let currentItem = links.find((link) =>
+    header?.classList.contains(link.className)
+  );
+  let currentLabel = currentItem?.label[lang] ?? links[0].label[lang];
```
Comment (jhsware) [inline kpm/kpm-backend/src/widget.js/notLoggedIn.js:74]: Should we have a fallback?

**Review by jhsware — CHANGES_REQUESTED** [review]
I have some minor questions about the code. See comments.

File: kpm/kpm-frontend/src/components/entrances.tsx [hunk 4]
```diff
@@ -4,16 +4,16 @@ import "./entrances.scss";
...
-  student: i18n("shortcut.studentweb"),
-};
+  intranet: i18n("shortcut.intranet"),
+  "student-web": i18n("shortcut.studentweb"),
+} as const;
 
 /** Get the current site */
-function getCurrentSite(): "external" | "intra" | "student" | undefined {
-  const SITES = ["external", "intra", "student"] as const;
+function getCurrentSite(): keyof typeof labels | undefined {
```
Comment (jhsware) [inline kpm/kpm-frontend/src/components/entrances.tsx:12]: Snyggt!

**Review by jhsware — APPROVED** [review]
LGTM

NOTE: I haven't test run the code.

**Pull request merged**